collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploaderTask.java

```diff
@@ -78,27 +79,23 @@ protected Outcome doInBackground(Long... instanceIdsToUpload) {
 
             String urlString = uploader.getUrlToSubmitTo(instance, deviceId, completeDestinationUrl);
 
-            SubmissionUploadResult result = uploader.uploadOneSubmission(instance, urlString, uriRemap);
-
-            // Don't add the instance that caused an auth request to the map because we want to
-            // retry; items in the map are not reconsidered in this submission attempt
-            if (result instanceof SubmissionUploadAuthRequested) {
-                outcome.authRequestingServer = ((SubmissionUploadAuthRequested) result).getAuthRequestingServerUri();
-            } else {
-                outcome.messagesByInstanceId.put(instance.getDatabaseId().toString(), result.getDisplayMessage());
-            }
-
-            if (result.isFatalError()) {
-                return outcome;
-            }
-
-            if (result.isSuccess()) {
+            try {
+                String customMessage = uploader.uploadOneSubmission(instance, urlString, uriRemap);
+                outcome.messagesByInstanceId.put(instance.getDatabaseId().toString(),
+                        customMessage != null ? customMessage : Collect.getInstance().getString(R.string.success));
                 Collect.getInstance()
                         .getDefaultTracker()
                         .send(new HitBuilders.EventBuilder()
                                 .setCategory("Submission")
                                 .setAction("HTTP")
                                 .build());
+            } catch (UploadAuthRequestedException e) {
+                outcome.authRequestingServer = e.getSubmissionUri();
+                // Don't add the instance that caused an auth request to the map because we want to
+                // retry; items in the map are not reconsidered in this submission attempt
```

Comment:
Wording here is a little confusing; better would be e.g.

"items present in the map are considered already attempted and won't be tried again"